src/crypto/crypto_util.cc

```diff
@@ -539,14 +572,14 @@ EnginePointer LoadEngineById(const char* id, CryptoErrorVector* errors) {
     if (ERR_get_error() != 0) {
       errors->Capture();
     } else {
-      errors->push_back(std::string("Engine \"") + id + "\" was not found");
+      errors->Insert(NodeCryptoError::ENGINE_NOT_FOUND);
```

Comment:
Since the error codes are fixed, I couldn't think of any way of incorporating the passed `id` for this error. Any suggestions please?

Reply:
So... two ideas:

1. You could add a second (optional) parameter to `Insert` that specifies the exact error message
2. You could do something similar to #37598 and parameterize `Insert()`, i.e. turn it into a template function that takes the error code + a variadic list of extra arguments, along the lines of:

```c++
template <typename... Args>
void CryptoErrorStore::Insert(const NodeCryptoError error, Args&&... args) {
  std::string error_string;
  ... /* same as before, with "Engine %s was not found" as the entry for ENGINE_NOT_FOUND */
  errors.emplace_back(SPrintF(error_string, std::forward<Args>(args)...));
}

/* use as: */
errors->Insert(NodeCryptoError::ENGINE_NOT_FOUND, id);
```

Option 1 is a bit simpler, whereas option 2 would be a bit more flexible for future changes.